momentGW/pbc/rpa.py

```diff
@@ -0,0 +1,498 @@
+"""
+Construct RPA moments with periodic boundary conditions.
+"""
+
+import numpy as np
+from pyscf import lib
+
+from momentGW import mpi_helper, util
+from momentGW.pbc.tda import dTDA
+from momentGW.rpa import dRPA as MoldRPA
+
+# TODO: Check lack of Lai in the integrals
+
+
+class dRPA(MoldRPA, dTDA):
+    """
+    Compute the self-energy moments using dRPA and numerical integration
+    with periodic boundary conditions.
+
+    Parameters
+    ----------
+    gw : BaseKGW
+        GW object.
+    nmom_max : int
+        Maximum moment number to calculate.
+    Lpx : numpy.ndarray
+        Density-fitted ERI tensor, where the first two indices
+        enumerate the k-points (as a dict), the third index is the auxiliary
+        basis function index, and the fourth and fifth indices are
+        the MO and Green's function orbital indices, respectively.
+    integrals : KIntegrals
+        Density-fitted integrals.
+    mo_energy : dict, optional
+        Molecular orbital energies. Keys are "g" and "w" for the Green's
+        function and screened Coulomb interaction, respectively.
+        If `None`, use `gw.mo_energy` for both. Default value is `None`.
+    mo_occ : dict, optional
+        Molecular orbital occupancies. Keys are "g" and "w" for the
+        Green's function and screened Coulomb interaction, respectively.
+        If `None`, use `gw.mo_occ` for both. Default value is `None`.
+    """
+
+    def __init__(
+        self,
+        gw,
+        nmom_max,
+        integrals,
+        mo_energy=None,
+        mo_occ=None,
+    ):
+        """
+        Key quantities for the dRPA calculation. These are constructed in advance to
+        reduce the computational cost of the quadratures.
+
+        returns
+        ----------
+        diag_eri : numpy.ndarray
+            Diagonal of the ERIs for each k-point.
+        d : numpy.ndarray
+            Array of orbital energy differences for each k-point.
+        Liad : dict of numpy.ndarray
+            The (Nkpt, Nkpt)(aux, W occ, W vir) integral array multiplied by the orbital energy
+            differences.
+        Liadinv : dict of numpy.ndarray
+            The (Nkpt, Nkpt)(aux, W occ, W vir) integral array divided by the orbital energy
+            differences.
```

Comment:
Improve docstrings for integrals, i.e. "keys should be the pair of k-point indices, and values are the ... at that k-point pair"